barretenberg/cpp/src/barretenberg/crypto/keccak/keccak.test.cpp

```diff
@@ -0,0 +1,134 @@
+#include "keccak.hpp"
+#include <algorithm>
+#include <array>
+#include <gtest/gtest.h>
+#include <vector>
+
+struct KeccakF1600TestVector {
+    std::array<uint64_t, KECCAKF1600_LANES> input;
+    std::array<uint64_t, KECCAKF1600_LANES> expected;
+};
+
+struct Keccak256TestVector {
+    std::string input;
+    std::array<uint8_t, KECCAK256_OUTPUT_BYTES> output;
+};
+
+const std::vector<KeccakF1600TestVector> keccak_f1600_test_vectors = {
+    {
+        // Test vector 0: input state all zeros
```

Comment:
Can you add an explicit reference in this file to where these come from? I know its also in the README